File: max/Max.java
public class Max {
    public static void main(String[] args) {
        int[] myList = {3, 1, 2};
        int max = getMax(myList);
        System.out.println(max);
    }
    
    public static int getMax(int[] inputArray){ 
        int maxValue = inputArray[0]; 
        for(int i=1;i<inputArray.length;i++){ 
            if(inputArray[i] > maxValue){ 
                maxValue = inputArray[i]; 
            } 
        } 
        return maxValue; 
    }
}

File: examples/array/max_in_array/max.c
#include <stdio.h>

static int maxValue(const int* array, int size) {
    int i;
    int max;
    max = array[0];
    for (i = 1; i < size/sizeof(array[0]); ++i) {
        if (max < array[i]) {
            max = array[i];
        }
    }
    return max;
}

int main(void) {
    const int array[] = {3, 1, 2};
    printf("%i\n", maxValue(array, sizeof(array)) );
}